pkg/external_repos/introspect_test.go

```diff
@@ -230,10 +241,59 @@ func TestUpdateIntrospectionStatusMetadata(t *testing.T) {
 				LastIntrospectionSuccessTime: &timestamp,
 				LastIntrospectionError:       pointy.String(""),
 				Status:                       config.StatusValid,
+				PackageCount:                 *pointy.Int(100),
+			},
+		},
+		{
+			name: "Status remains as Invalid",
+			given: TestCaseGiven{
+				status: config.StatusInvalid,
+				count:  1,
+				err:    fmt.Errorf("Error remains, keep it as Invalid"),
+			},
+			expected: dao.Repository{
+				LastIntrospectionTime:        &timestamp,
+				LastIntrospectionUpdateTime:  nil,
+				LastIntrospectionSuccessTime: nil,
+				LastIntrospectionError:       pointy.String("Error remains, keep it as Invalid"),
+				Status:                       config.StatusInvalid,
+				PackageCount:                 *pointy.Int(100),
 			},
 		},
-		// Status is valid and count is 0
 		{
+			name: "Status remains as Unavailable",
+			given: TestCaseGiven{
+				status: config.StatusUnavailable,
+				count:  1,
+				err:    fmt.Errorf("Error ramins Unavailable"),
+			},
+			expected: dao.Repository{
+				LastIntrospectionTime:        &timestamp,
+				LastIntrospectionUpdateTime:  nil,
+				LastIntrospectionSuccessTime: nil,
+				LastIntrospectionError:       pointy.String("Error ramins Unavailable"),
+				Status:                       config.StatusUnavailable,
+				PackageCount:                 *pointy.Int(100),
+			},
+		},
+		{
+			name: "Status is set to Unavailable when error an any other current status",
```

Comment:
What if you rename this to "Status change to Unavailable" and make the TestCaseGiven status "AnythingExceptPending", then remove the test case for Valid to Unavailable (since this overlaps with that).

Reply:
thanks mate! I have tried it, but it leaves the line below not covered :

```golang
	case config.StatusValid:
		output.Status = config.StatusUnavailable // Repository introspected successfully at least once, but now errors
```

so leaving as it is.

Reply:
oh I see! It's testing the default case